js/skin.js

```diff
@@ -340,268 +384,338 @@ const Skin = createReactClass({
-          screen = <Spinner loadingImage={this.props.skinConfig.general.loadingImage.imageResource.url} />;
-          break;
-        case CONSTANTS.SCREEN.START_SCREEN:
-          screen = (
-            <StartScreen
-              {...this.props}
-              componentWidth={this.state.componentWidth}
-              contentTree={this.state.contentTree}
-              isInitializing={false}
-            />
-          );
-          break;
-        case CONSTANTS.SCREEN.START_LOADING_SCREEN:
-          screen = (
-            <StartScreen
-              {...this.props}
-              componentWidth={this.state.componentWidth}
-              contentTree={this.state.contentTree}
-              isInitializing={false}
-              showSpinner={true}
-            />
-          );
-          break;
-        case CONSTANTS.SCREEN.PLAYING_SCREEN:
-          screen = (
-            <PlayingScreenWithAutoHide
-              {...this.props}
-              handleVrPlayerMouseDown={this.handleVrPlayerMouseDown}
-              handleVrPlayerMouseMove={this.handleVrPlayerMouseMove}
-              handleVrPlayerMouseUp={this.handleVrPlayerMouseUp}
-              handleVrPlayerClick={this.handleVrPlayerClick}
-              handleVrPlayerFocus={this.handleVrPlayerFocus}
-              handleTouchEnd={this.handleTouchEnd}
-              isVrMouseMove={this.state.isVrMouseMove}
-              contentTree={this.state.contentTree}
-              currentPlayhead={this.state.currentPlayhead}
-              duration={this.state.duration}
-              buffered={this.state.buffered}
-              fullscreen={this.state.fullscreen}
-              playerState={this.state.playerState}
-              seeking={this.state.seeking}
-              upNextInfo={this.state.upNextInfo}
-              isLiveStream={this.state.isLiveStream}
-              responsiveView={this.state.responsiveId}
-              componentWidth={this.state.componentWidth}
-              componentHeight={this.state.componentHeight}
-              videoQualityOptions={this.state.videoQualityOptions}
-              closedCaptionOptions={this.props.closedCaptionOptions}
-              captionDirection={this.props.controller.captionDirection}
-              ref="playScreen"
-            />
-          );
-          break;
-        case CONSTANTS.SCREEN.SHARE_SCREEN:
-          screen = (
-            <ContentScreen {...this.props} screen={CONSTANTS.SCREEN.SHARE_SCREEN} icon="share">
-              <SharePanel
+      if (this.props.controller.state.audioOnly) {
+        switch (this.state.screenToShow) {
+          case CONSTANTS.SCREEN.INITIAL_SCREEN:
+          case CONSTANTS.SCREEN.START_SCREEN:
+          case CONSTANTS.SCREEN.PLAYING_SCREEN:
+          case CONSTANTS.SCREEN.PAUSE_SCREEN:
+          case CONSTANTS.SCREEN.END_SCREEN:
+            screen = (
+              <AudioOnlyScreen
+                {...this.props}
+                handleVrPlayerMouseDown={this.handleVrPlayerMouseDown}
```

Comment:
Should we ignore or disable these VR events for audio only?

Reply:
Sounds good, will add